server/sockets/utilityFunctions.js

```diff
@@ -0,0 +1,21 @@
+// socket utility functions
+const db = require('../../database/database.js');
+
+
+module.exports = {
+  dropNCreateDBTables: () => {
+    db.complex.dropAllTables()
+    .then(db.complex.initializeDB)
+    .catch(err => console.error('Error dropping or creating tables', err));
```

Comment:
I'm curious about the dropNCreateDB function.  Previously we had it running when the server started, but now it runs when the socket connects (unless I'm misunderstanding).  What implications will this have for testing, given what we just learned about beforeAll()?